cmd/kube-proxy/app/server.go

```diff
@@ -203,6 +204,7 @@ func (o *Options) AddFlags(fs *pflag.FlagSet) {
 	fs.Var(&o.config.DetectLocalMode, "detect-local-mode", "Mode to use to detect local traffic. This parameter is ignored if a config file is specified by --config.")
 	fs.StringVar(&o.config.DetectLocal.BridgeInterface, "pod-bridge-interface", o.config.DetectLocal.BridgeInterface, "A bridge interface name in the cluster. Kube-proxy considers traffic as local if originating from an interface which matches the value. This argument should be set if DetectLocalMode is set to BridgeInterface.")
 	fs.StringVar(&o.config.DetectLocal.InterfaceNamePrefix, "pod-interface-name-prefix", o.config.DetectLocal.InterfaceNamePrefix, "An interface prefix in the cluster. Kube-proxy considers traffic as local if originating from interfaces that match the given prefix. This argument should be set if DetectLocalMode is set to InterfaceNamePrefix.")
+	logsapi.AddFlags(&o.config.Logging, fs)
```

Comment:
Aren't all command line flags already deprecated in favor of the config fields?

```console
$ go run ./cmd/kube-proxy --logging-format=json
I0510 15:26:19.269444   67867 server.go:235] "Warning, all flags other than --config, --write-config-to, and --cleanup are deprecated, please begin using a config file ASAP"
I0510 15:26:19.269491   67867 server_others.go:75] "Using iptables proxy"
{"ts":1683725179269.8264,"caller":"app/server.go:538","msg":"Neither kubeconfig file nor master URL was specified, falling back to in-cluster config","v":0}
{"ts":1683725179269.8413,"caller":"app/server.go:491","msg":"Error running ProxyServer","err":"unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"}
{"ts":1683725179269.864,"caller":"cli/run.go:74","msg":"command failed","err":"unable to load in-cluster configuration, KUBERNETES_SERVICE_HOST and KUBERNETES_SERVICE_PORT must be defined"}
exit status 1
```

With that in mind, does it make sense to register these flags?

Reply:
Another observation: ideally, the code which runs before `ValidateAndApplyAsField` shouldn't log anything because the desired logging configuration is still unknown. That's why we have a mixture of text and JSON log output here.

Reply:
We've been kind of terrible about actually deprecating the command-line options in favor of config files, but yes, you're right, we should say that if people want to use the cool new features, they have to use the config file

Reply:
On the other hand, these flags are the same that also other commands have. For the sake of consistency it would be nice if they were supported also in kube-proxy, not just the config options.

But then the warning needs to be modified so that it doesn't trigger for the logging parameters.

Reply:
ref https://github.com/kubernetes/kubernetes/pull/101273#issuecomment-823357862

Reply:
In the latest changes, I kept the logging-format flag and modified the warning, making sure the code which runs before ValidateAndApplyAsField shouldn't log anything.

Please review the changes, Thanks.